robot/impl/resource_manager_test.go

```diff
@@ -1850,3 +1852,76 @@ func managerForDummyRobot(t *testing.T, robot robot.Robot) *resourceManager {
 	}
 	return manager
 }
+
+// TestReconfigureParity validates that calling the synchronous and asynchronous version
+// of `Reconfigure` results in the same resource state on a local robot for all
+// combinations of an initial and updated configuration.
+func TestReconfigureParity(t *testing.T) {
```

Comment:
let me know if this is either overkill or missing some important use cases

Reply:
this is probably overkill, I think being making the tests smaller but more targeted will be easier to maintain later on

Reply:
unfortunately this test is quite cumbersome to extend for configs with modules, which is what we really care about